dsc_lib/src/functions/contains.rs

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use crate::DscError;
+use crate::configure::context::Context;
+use crate::functions::{AcceptedArgKind, Function, FunctionCategory};
+use rust_i18n::t;
+use serde_json::Value;
+use tracing::debug;
+
+#[derive(Debug, Default)]
+pub struct Contains {}
+
+impl Function for Contains {
+    fn description(&self) -> String {
+        t!("functions.contains.description").to_string()
+    }
+
+    fn category(&self) -> FunctionCategory {
+        FunctionCategory::Array
+    }
+
+    fn min_args(&self) -> usize {
+        2
+    }
+
+    fn max_args(&self) -> usize {
+        2
+    }
+
+    fn accepted_arg_types(&self) -> Vec<AcceptedArgKind> {
+        vec![AcceptedArgKind::Array, AcceptedArgKind::Object, AcceptedArgKind::String, AcceptedArgKind::Number]
+    }
+
+    fn invoke(&self, args: &[Value], _context: &Context) -> Result<Value, DscError> {
+        debug!("{}", t!("functions.contains.invoked"));
+        let mut found = false;
+
+        let (string_to_find, number_to_find) = if let Some(string) = args[1].as_str() {
+            (string.to_string(), 0)
```

Comment:
I think assigning `number_to_find` to 0 here could potentially allow for an edge case when the container is an array of integers

Reply:
Changed logic to use `None` and added a test case for your scenario